processing-service-core/api/swagger/swagger.yaml

```diff
@@ -952,7 +954,246 @@ paths:
+        description: The key to use when searching for a tenants configuration.
+        required: true
+    get:
+      tags:
+        - Tenant
+      summary: Retrieve effective tenant specific config.
+      description: Returns the effective tenant specific config for this key. That can be either a custom value specified for the tenant or the default.
+      operationId: getEffectiveTenantConfig
+      responses:
+        200:
+          description: Returns the effective tenant configurations for this key.
+          schema:
+            $ref: "#/definitions/effectiveTenantConfig"
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+  #
+  #
+  /tenants/{tenantId}/effectiveconfig/:
+    x-swagger-router-controller: tenant
+    parameters:
+      - name: tenantId
+        in: path
+        type: string
+        description: The ID tenant.
+        required: true
+    get:
+      tags:
+        - Tenant
+      summary: Retrieve effective tenant specific config.
+      description: Returns the effective tenant specific config for this key. That can be either a custom value specified for the tenant or the default.
+      operationId: getEffectiveTenantConfigs
+      responses:
+        200:
+          description: Returns all effective configurations for this tenant.
+          schema:
+            $ref: "#/definitions/effectiveTenantConfigs"
+        400:
+          description: Indicates invalid parameter was passed.
+          schema:
+            $ref: "#/definitions/errorMessage"
+        404:
+          description: Indicates key was not found.
+  #
+  #
 definitions:
+  tenantConfigs:
+    type: object
+    properties:
+      tenantId:
+        type: string
+        description: The tenantId used in retrieval of the custom configurations.
+      configurations:
+        type: array
+        items:
+          $ref: "#/definitions/tenantConfig"
+        description: The collection of custom configurations set for the supplied tenantId.
+  tenantConfig:
+    type: object
+    properties:
+      tenantId:
+        type: string
+        description: The tenantId that was used in retrieving the values.
+      key:
+        type: string
+        description: The key that used in retrieval of the custom config.
+      value:
+        type: string
+        description: The value stored for the custom config key.
+  effectiveTenantConfigs:
+    type: object
+    properties:
+      tenantId:
+        type: string
+        description: The tenantId used in retrieval of the custom configurations.
+      configurations:
+        type: array
+        items:
+          $ref: "#/definitions/effectiveTenantConfig"
+        description: The collection of custom configurations set for the supplied tenantId.
+  effectiveTenantConfig:
+    type: object
+    properties:
+      tenantId:
```

Comment:
Doesn't having this here just mean that you are constantly repeating back something you were given in the request in the first place?